esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -770,13 +795,43 @@ class CEsdlSDSStore : implements IEsdlStore, public CInterface
         esdldeftree->setProp("@id", lcId.str());
         esdldeftree->setProp("@esdlservice", esdlServiceName);
 
-        if (methodsConfig != nullptr)
-            esdldeftree->addPropTree("Methods", LINK(methodsConfig));
+#define IDENTIFY_ESDL_ENTRY(node, root, child) \
```

Comment:
Wouldn't this be more clear as a function.   And a name that was more self documenting would be nice. For example, passing methodsConfig as a reference?:

IPTree *srcBinding = walkEsdlConfigLevel(methodsConfig, ESDL_BINDING_ENTRY, ESDL_DEF_ENTRY);

I would imediately realize you were walking the levels and potentially updating methodsConfig as you went.

Name could be something else, but meaning that it is walking or checking each potential level.
checkEsdlConfigLevel? getEsdlConfigLevel?

Reply:
And the last one would be clearest as just srcMethods = methodsConfig;

Reply:
I chose a macro because it simplified the code that used it, compared to making function calls.
* It handles the declaration of the variable in the calling environment, which may or may not be a good thing.
* It returns errors from the calling environment without needing two return values (or an additional reference parameter) from a function.
* It doesn't require the function to take extra parameters just to prepare the error message, or require me to replicate error message preparation in the caller. Replication can probably be handled by a macro.

I also reused the macro for Methods because I wasn't sure if we would ever want to enforce that we actually have a Methods element, or will always assume that the user of the store always sends either Binding, Definition, or Methods.

I'll refactor.